### DISCUSSION TITLE
How to trigger replication

### DISCUSSION THREAD

**Discussion by halkeye** [longhorn/longhorn#3480] [Q&A] [answered]
I have a mini 3 node k8s cluster at home to play around with. This weekend I decided to physically move some nodes around which caused some syncing issues. I let things catch back up, and upgraded from 1.1.1 to 1.2.3.

I should have all my volumes set to 3 (2 if the volume is old enough) replicas. But I have 1 node with 0 replicas, one with 3, and one with 25 replicas for the total of 28 volumes. Which essentially means I have 0 replicas.

My server is kubeadm based, running
```
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.6", GitCommit:"8a62859e515889f07e3e3be6a1080413f17cf2c3", GitTreeState:"clean", BuildDate:"2021-04-15T03:19:55Z", GoVersion:"go1.15.10", Compiler:"gc", Platform:"linux/amd64"}
```

I haven't yet been able to find any errors in the kubectl logs, and all of the volumes themselves don't show any events from the other nodes. So I'm at a bit of a loss what to do next, or how to trigger things for a specific volume so I can watch logs.

As far as I can tell everything else is green.

![image](https://user-images.githubusercontent.com/110087/147867689-bb090041-60d1-4a3b-8172-9fdb1e0476f3.png)

Current replica count is 2, but it only seems to be replicated once, unless I misunderstand - 
![image](https://user-images.githubusercontent.com/110087/147992791-b194124e-a0d7-4cab-8b31-bb62f16d15be.png)

using `stern -n longhorn-system . --tail=1` to look at all logs the only error I'm setting at all is
```
csi-snapshotter-645cf6bd97-plzkv csi-snapshotter E0104 00:28:40.788621       1 reflector.go:127] github.com/kubernetes-csi/external-snapshotter/client/v3/informers/externalversions/factory.go:117: Failed to watch *v1beta1.VolumeSnapshotClass: failed to list *v1beta1.VolumeSnapshotClass: the server could not find the requested resource (get volumesnapshotclasses.snapshot.storage.k8s.io)
```

**Comment by jenting** [comment, marked as answer]
The replicas are schedule to 2 nodes.
Since now you have 3 healthy and ready nodes and I guess you want the replicas be balanced across the nodes? If yes, you could check the setting replica auto rebalance. Or, you could manually delete the replicas on the node with 25 replicas, the longhorn scheduler will schedule the replica by scheduling algorithm.

**Reply by halkeye** [reply]
if i delete the only replica, won't it delete the data?

**Reply by halkeye** [reply]
(you said "Oops, sorry I didn't realized you only have 1 replica per volume") but I wanted to fix the threading.

**Reply by jenting** [reply]
Could you please check Node page to see if the nodes with available space for the replicas could be scheduled?
Also, could you please check the settings `replica-soft-anti-affinity` and `replica-zone-soft-anti-affinity`?

**Reply by halkeye** [reply]
![image](https://user-images.githubusercontent.com/110087/147995539-ffbdecb9-de0b-49d6-86f2-0bb4b47774ab.png)

![image](https://user-images.githubusercontent.com/110087/147995575-4af7f0f7-2fd9-4bd3-80c2-f1a5c3739edd.png)


yep yep, i was poking around with the settings after it didn't replicate, but i think i left thme in a good state.

**Reply by jenting** [reply]
Could you please send the support bundle to longhorn-support-bundle@suse.com?

We'd like to check the following settings as well as the logs
- replica replenishment wait interval.
- concurrent replica rebuilding per-node limit.

**Reply by halkeye** [reply]
2.5mb sent. hopefully email will allow that big of attachment

**Reply by halkeye** [reply]
I was poking at the logs it generated and found
```
2022-01-03T02:28:39.940830651Z time="2022-01-03T02:28:39Z" level=info msg="cannot list the content of the src directory /var/lib/rancher/longhorn/engine-binaries for the copy, will do nothing: Failed to execute: nsenter [--mount=/host/proc/1/ns/mnt --net=/host/proc/1/ns/net bash -c ls /var/lib/rancher/longhorn/engine-binaries/*], output , stderr, ls: cannot access '/var/lib/rancher/longhorn/engine-binaries/*': No such file or directory\n, error exit status 2"
``` 
in longhorn-manager, is that bad?

**Reply by jenting** [reply]
You configure the `Concurrent Replica Rebuild Per Node Limit` to `0`. The `0` means disabling the replica rebuilding.
Could you please increase the value?

Thanks @PhanLe1010 help :pray:

**Reply by halkeye** [reply]
oh i see it in the danger zone. I set it to 2. What would be the default? I don't think I've ever intentionally set that value.

omg omg omg omg, the number of healthy volumes are going up.

**Reply by jenting** [reply]
> oh i see it in the danger zone. I set it to 2. What would be the default? I don't think I've ever intentionally set that value.

The default value is 5 https://github.com/longhorn/longhorn-manager/blob/5b64adfaa300d2765cfeba232b883cf35b28e1bc/types/setting.go#L615.

> omg omg omg omg, the number of healthy volumes are going up.

:rocket:

**Comment by halkeye** [comment]


**Reply by jenting** [reply]
Oops, sorry I didn't realized you only have 1 replica per volume
